File: Delivery/src/main/java/fooddelivery/infra/PolicyHandler.java
package fooddelivery.infra;

import javax.naming.NameParser;

import javax.naming.NameParser;
import javax.transaction.Transactional;

import fooddelivery.config.kafka.KafkaProcessor;
import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.cloud.stream.annotation.StreamListener;
import org.springframework.messaging.handler.annotation.Payload;
import org.springframework.stereotype.Service;
import fooddelivery.domain.*;


@Service
@Transactional
public class PolicyHandler{
    @Autowired 배달관리Repository 배달관리Repository;
    
    @StreamListener(KafkaProcessor.INPUT)
    public void whatever(@Payload String eventString){}

    @StreamListener(value=KafkaProcessor.INPUT, condition="headers['type']=='조리완료됨'")
    public void whenever조리완료됨_배달대상정보받음(@Payload 조리완료됨 조리완료됨){

        조리완료됨 event = 조리완료됨;
        System.out.println("\n\n##### listener 배달대상정보받음 : " + 조리완료됨 + "\n\n");


        

        // Sample Logic //

        

    }

}

File: app/PolicyHandler.py
import util
import 주문DB
from 주문 import 주문
주문repository = 주문DB.repository

import 주문DB
from 주문 import 주문
주문repository = 주문DB.repository


from 배달시작됨 import 배달시작됨

def whenever배달시작됨_주문상태변경(data):
    event = 배달시작됨()
    event = util.AutoBinding(data, event)
    
    주문 = 주문()
    주문repository.save(주문)
    주문 = 주문()
    주문repository.save(주문)
    
from 배달시작됨 import 배달시작됨

def whenever배달시작됨_주문상태변경(data):
    event = 배달시작됨()
    event = util.AutoBinding(data, event)
    
    주문 = 주문()
    주문repository.save(주문)
    주문 = 주문()
    주문repository.save(주문)
    

